### ISSUE TITLE
INCOLLECTION entry type produces wrong results without editor field

### ISSUE THREAD

**Issue by Witiko** [michal-h21/biblatex-iso690#68]
Create a LaTeX document named “example.tex” containing the following MWE:

```tex
\documentclass{article}
\usepackage{filecontents}
\begin{filecontents}{example.bib}
@INCOLLECTION{t02,
  author = {Greenberg, David},
  year = {1998},
  title = {Camel drivers and gatecrashers},
  subtitle = {quality control in the digital research library},
  editor = {Hawkins, B.L and Battin, P},
  booktitle = {The mirage of continuity},
  booksubtitle = {reconfiguring academic information
                  resources for the 21st century},
  location = {Washington (D.C.)},
  publisher = {Council on Library and Information Resources;
               Association of American Universities}, 
  pages = {105-116},
  langid = {english},
}
\end{filecontents}
\usepackage[style=iso-authoryear]{biblatex}
\addbibresource{example.bib}
\begin{document}
\nocite{*}
\printbibliography
\end{document}
```

Typesetting the document using `latexmk -pdf example` produces the following output on TeX Live 2018:

![example.pdf preview before removing the editor field](https://user-images.githubusercontent.com/603082/47956902-edb17880-dfac-11e8-9600-d0b610689cc6.png)

Now remove the `editor` field as follows and typeset the document again using `latexmk -pdf example`:

```diff
9d8
<   editor = {Hawkins, B.L and Battin, P},
```

This produces the following output on TeX Live 2018:

![example.pdf preview after removing the editor field, actual output](https://user-images.githubusercontent.com/603082/47956924-64e70c80-dfad-11e8-8cd4-eb21814a4f53.png)

In the absence of the `editor` field, the title of the collection entry is used. The expected behavior is that only the collection title and subtitle are displayed as follows:

![example.pdf preview after removing the editor field, expected output](https://user-images.githubusercontent.com/603082/47957008-aa580980-dfae-11e8-9eac-80f4f23ee41a.png)

**Comment by moewew** [comment]
This should be the same general issue as https://github.com/michal-h21/biblatex-iso690/issues/58 (and also https://github.com/michal-h21/biblatex-iso690/issues/57 which was resolved differently). The `@incollection` driver should not be using `\usebibmacro{editor}`, instead it should be using a safe version that is not redefined `authoryear`-like styles.

**Comment by Witiko** [comment]
You are right. I originally thought the issue was related to the use of the `crossref` field, so I did not find a related issue before deciding to open a new one. Leaving this open, so that a potential future contributor is motivated by the thought of closing two issues in a single commit.

**Comment by michal-h21** [comment]
I've renamed the [editor macro](https://github.com/michal-h21/biblatex-iso690/blob/75fb6c3edeb2953f09922e19f141e3025a94794b/iso.bbx#L238) to `incollection-editor`. It seems to fix the issue.

**Comment by Witiko** [comment]
Thank you for your fast response. Barring any regressions, commit 75fb6c3 fixes this issue.

**Comment by moewew** [comment]
This introduces a regression for `iso-numeric` and `iso-alphabetic`
```
\documentclass{article}
\usepackage{filecontents}

\usepackage[style=iso-numeric]{biblatex}
\addbibresource{biblatex-examples.bib}
\begin{document}
\nocite{westfahl:frontier}
\printbibliography
\end{document}
```
Before the fix the output was
> WESTFAHL, Gary **(ed.)**. Space and Beyond: The Frontier Theme in Science Fiction. Space and Beyond: The Frontier Theme in Science Fiction. Westport, Conn. and London: Greenwood, 2000.

now it is
> WESTFAHL, Gary<b>, ed.</b> Space and Beyond: The Frontier Theme in Science Fiction. Space and Beyond: The Frontier Theme in Science Fiction. Westport, Conn. and London: Greenwood, 2000.

That is because `iso-numeric` and `iso-alphabetic` now take their `editor` macro from the standard styles and not from `iso.bbx` , which does not redefine `editor` any more. `iso-authoryear` and `iso-authortitle` still redefine `editor` and thus still give the expected output.

If I were to write the style now I would not redefine the `editor` macro at all. Instead I would be using the definitions from the corresponding standard styles and
```
\DeclareFieldFormat{editortype}{\mkbibparens{#1}} 
\DeclareDelimFormat{editortypedelim}{\addspace}
```
possibly the same for the `translator`. Together with the option `dashed=false` the output should be as expected.

Additionally, I'm not sure about the use of `\ifuseeditor` in `incollection-editor`, but that might be a judgement call.

**Comment by michal-h21** [comment]
@moewew thanks for catching this. I've removed the `editor` redefinitions and added the formats you provided and it seems to work as expected, even without the `dashed=false` option.

**Comment by moewew** [comment]
Note that `editortype` and `editortypedelim` were only introduced fairly recently in version 3.8 of `biblatex`. I wasn't aware of the project stance on backwards compatibility (to me it seems that there is still some older code hanging around for backwards compatibility reasons), so I did not suggest a pull request myself (and only wrote how I would handle things today).

**Comment by michal-h21** [comment]
I think it is OK to get rid of the old code, there could be regressions with every new Biblatex release otherwise. The last release of the package on CTAN was year and half ago, so it would be nice to do some cleanup before new release.

**Comment by DavidLuptak** [comment]
This was a tricky one. The removal of redefinition of `editor` macro in `iso-authoryear.bbx` 0e4eded3d7182d989b3d8cae7d142bbd9841e4aa introduced a regression in `iso-authoryear` style, because, unfortunately, the macro `labeltitle` that handled printing label or title in place of author or editor names if they are unavailable, was executed from this `editor` macro. That's why we didn't get the desired results, e.g. for bib entry "[cs](https://github.com/michal-h21/biblatex-iso690/blob/master/mybib.bib#L135-L144)" we got
```
1991. Titulek. The journal of big papers. Vol. 12, no. 2, pp. 22–44. Available from DOI: 10.1000/182.
```
instead of
```
Titulek, 1991. The journal of big papers. Vol. 12, no. 2, pp. 22–44. Available from DOI: 10.1000/182.
```

I redefined the `names:primary` macro in `iso-authoryear.bbx` to meet all the requirements that we faced in c815202.

If there are no more pitfalls, this issue could be closed.
